ext/dom/dom.stub.php

```diff
@@ -0,0 +1,313 @@
+
+    /** @return bool */
+    public function schemaValidateSource(string $source, int $flags = 0) {}
+
+    /** @return bool */
+    public function validate() {}
+
+    /** @return int|false */
+    public function xinclude(int $options = 0) {}
+
+    /** @return DOMNode|false */
+    public function adoptNode(DOMNode $source) {}
+}
+
+class DOMDocumentFragment {
+    public function __construct() {}
+
+    /** @return bool */
+    public function appendXML(string $data) {}
+}
+
+class DOMElement {
+    public function __construct(string $name, ?string $value = null, ?string $uri = null) {}
+
+    /** @return string */
+    public function getAttribute(string $name) {}
+
+    /** @return string */
+    public function getAttributeNS(?string $namespaceURI, string $localName) {}
+
+    /** @return DOMAttr|DOMNamespaceNode|null */
+    public function getAttributeNode(string $name) {}
+
+    /** @return DOMAttr|DOMNamespaceNode|null */
+    public function getAttributeNodeNS(?string $namespaceURI, string $localName) {}
+
+    /** @return DOMNodeList */
+    public function getElementsByTagName(string $name) {}
+
+    /** @return DOMNodeList */
+    public function getElementsByTagNameNS(string $namespaceURI, string $localName) {}
+
+    /** @return bool */
+    public function hasAttribute(string $name) {}
+
+    /** @return bool */
+    public function hasAttributeNS(?string $namespaceURI, string $localName) {}
+
+    /** @return bool */
+    public function removeAttribute(string $name) {}
+
+    /** @return DOMAttr|false */
+    public function removeAttributeNS(?string $namespaceURI, string $localName) {}
+
+    /** @return DOMAttr|false */
+    public function removeAttributeNode(string $name) {}
+
+    /** @return DOMAttr|bool */
+    public function setAttribute(string $name, string $value) {}
+
+    /** @return null|false */
+    public function setAttributeNS(?string $namespaceURI, string $localName, string $value) {}
+
+    /** @return DOMAttr|null|false */
+    public function setAttributeNode(DOMAttr $attr) {}
+
+    /** @return DOMAttr|null|false */
+    public function setAttributeNodeNS(DOMAttr $attr) {}
+
+    public function setIdAttribute(string $name, bool $isId) {}
+
+    public function setIdAttributeNS(string $namespaceURI, string $localName, bool $isId) {}
+
+    public function setIdAttributeNode(DOMAttr $attr, bool $isId) {}
+}
+
+class DOMEntityReference {
+    public function __construct(string $name) {}
+}
+
+class DOMImplementation {
+    public function getFeature(string $feature, string $version) {}
+
+    /** @return bool */
+    public function hasFeature(string $feature, string $version) {}
+
+    public function createDocumentType($qualifiedName, $publicId, $systemId) {}
+
+    /** @return DOMDocument|false */
+    public function createDocument(?string $namespaceURI = null, ?string $qualifiedName = null, ?DOMDocumentType $doctype = null) {}
```

Comment:
```suggestion
    public function createDocument(string $namespaceURI = "", string $qualifiedName = "", DOMDocumentType $doctype = UNKNOWN) {}
```

Though you might want to make `$doctype` nullable in implementation.